crates/tako/src/internal/server/workerload.rs

```diff
@@ -246,16 +249,20 @@ impl WorkerLoad {
     pub(crate) fn load_wrt_request(&self, wr: &WorkerResources, request: &ResourceRequest) -> u32 {
         let mut result = 0;
         for entry in request.entries() {
-            let count = wr.n_resources.get(entry.resource_id).copied().unwrap_or(0);
-            if count == 0 {
+            let count = wr
+                .n_resources
+                .get(entry.resource_id)
+                .copied()
+                .unwrap_or(ResourceAmount::ZERO);
+            if count.is_zero() {
                 return 0;
             }
             let load = self
                 .n_resources
                 .get(entry.resource_id)
                 .copied()
-                .unwrap_or(0);
-            result += ((load * 512) / (count * 512)) as u32;
+                .unwrap_or(ResourceAmount::ZERO);
+            result += (load.total_fractions() * 512 / count.total_fractions()) as u32;
```

Comment:
It would be possible but here it is more like recomputing to a different precision fixed point number. So dividing resource by resource would be slightly consfusing (maybe).